Library/Homebrew/extend/os/mac/unpack_strategy/zip.rb

```diff
@@ -2,12 +2,23 @@ module UnpackStrategy
   class Zip
     prepend Module.new {
       def extract_to_dir(unpack_dir, basename:, verbose:)
-        volumes = super.stderr.chomp
-                       .split("\n")
-                       .map { |l| l[/\A   skipping: (.+)  volume label\Z/, 1] }
-                       .compact
+        result = super
 
-        return if volumes.empty?
+        volumes = result.stderr.chomp
+                        .split("\n")
+                        .map { |l| l[/\A   skipping: (.+)  volume label\Z/, 1] }
+                        .compact
+
+        if volumes.empty?
+          # Merge ._ files back into native, extended attributes.
+          if result.stdout.lines.grep(/\._/)
+            system_command!("dot_clean",
+                            args: ["-mnv", "--keep=dotbar", unpack_dir],
```

Comment:
```
     -m      Always delete dot underbar files.
     -n      Delete dot underbar file if there is no matching native file.
```

Does `-m` not imply `-n`? Also, `-v` should only be passed if `verbose` is true.